codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java

```diff
@@ -456,24 +520,31 @@ public ChannelFuture writeWindowUpdate(ChannelHandlerContext ctx, int streamId,
     @Override
     public ChannelFuture writeFrame(ChannelHandlerContext ctx, byte frameType, int streamId,
```

Comment:
@normanmaurer TBH I found a LOT of place where this optmization could be applied, but the ones where I place it are the ones which popup in my profiling data. 

Maybe  @bryce-anderson or @idelpivnitskiy could spot other important points which could be worthy to fix, because still hot for other common cases

eg everytime we allocate just a  bunch of bytes and we expect to write small payloads, actually